### DISCUSSION TITLE
One To Many (Latest)

### DISCUSSION THREAD

**Discussion by cheadevit** [Laravel-Backpack/community-forum#607] [Q&A (Help)] [answered]
Hello Guys. 

I have a database that contains the table as below, 

![Screenshot 2023-08-02 at 15 45 15](https://github.com/Laravel-Backpack/community-forum/assets/52439329/06470db5-0000-4e1b-b8c0-60d4cc0c970e)

in case 1 Asset has many conditions, example asset_1 can have asset_con_1, asset_con_2, asset_con_3 and etc...
and the problem is when I tried to use the field type `relationship`, the field showed select2_muliple

here is the model function

```
public function conditionHasOne()
    {
        return $this->belongsToMany(Condition::class, 'asset_conditions', 'asset_id', 'condition_id')->orderBy('asset_conditions.id', 'desc')->limit(1)->withTimestamps();
    }
```
here is the field in the update operation

```
$this->crud->addField([
            'label'     => 'Condition',
            'type'      => 'relationship',
            'name'      => 'condition_id',
            'entity'    => 'conditionHasOne',
            'attribute' => 'name',
            'wrapper' => [
                'class' => 'form-group col-md-6'
            ],
            'model'     => "App\Models\Condition",
            'options'   => (function ($query) {
                return $query->where('lookup_type_id', LookupConstant::ASSET_CONDITION)->get();
            }),
        ]);
```

here is the result

![Screenshot 2023-08-02 at 15 22 05](https://github.com/Laravel-Backpack/community-forum/assets/52439329/f82bf148-ca31-4d6a-8a26-1a5c3b1ad8ae)

The question is how to achieve this UI

![Screenshot 2023-08-02 at 15 24 53](https://github.com/Laravel-Backpack/community-forum/assets/52439329/1736a7f4-5de8-4095-942a-717afb697366)

**Comment by pxpm** [comment, marked as answer]
Hey @cheadevit 

The field shows multiple because ... it's a multiple relation. (BelongsTo**Many**)

You can try `multiple => false` in your field, but not sure if you don't need to overwrite the saving process because Backpack is expecting an array for a multiple relation, even if it's an array with only one element. 

Let me know how it goes.

Cheers

**Reply by cheadevit** [reply]
@pxpm it works now, I appreciated it.
Thank you.

Cheers

**Reply by cheadevit** [reply]
@here is the code

```
$this->crud->addField([
            'label'     => 'Condition',
            'type'      => 'relationship',
            'name'      => 'condition_id',
            'entity'    => 'conditionLatest',
            'attribute' => 'name',
            'wrapper'   => [
                'class' => 'form-group col-md-6'
            ],
            'multiple'  => false,
            'model'     => "App\Models\Condition",
            'options'   => (function ($query) {
                return $query->where('lookup_type_id', LookupConstant::ASSET_CONDITION)->get();
            }),
       ]);
```

**Reply by pxpm** [reply]
Thanks for sharing back your solution, it will help others in future with the same problem. 🙏 

Cheers
